doc.go

```diff
@@ -147,9 +147,15 @@
 //      CheckOrigin: func(r *http.Request) bool { return true },
 //  }
 //
-// The deprecated Upgrade function does not enforce an origin policy. It's the
-// application's responsibility to check the Origin header before calling
-// Upgrade.
+// We recommend the upgrader.Upgrade method to perform an upgrade
+// from an HTTP connection to a websocket connection. This method performs
```

Comment:
You're right. I don't see the harm, though, in including this in the Overview section. You could stipulate that it's redundant; I'm not arguing with that. But redundancy isn't always harmful and I think, in this case, this is good information to include in the Overview. It clarifies the distinction between the Upgrader.Upgrade method and the package-level Upgrade function immediately, setting the user on the right path.

Personally, I found myself scrolling between the overview section, the documentation for the package-level Upgrade function and the Upgrader.Upgrade method to understand the difference between the two. Now, it's obvious to me what that difference is. But, having this nota bene in the Overview section would have saved me some time.

I don't expect you to find this addition that useful (as the package's primary author). But, I believe that it could prove useful to other users in the future and I don't imagine that adding it would cause universal frustration.